worlds/shapez/locations.py

```diff
@@ -0,0 +1,578 @@
+    "Level 1000": "TODO",
+    "Belt Upgrade Tier II": "TODO",
+    "Miner Upgrade Tier II": "TODO",
+    "Processors Upgrade Tier II": "TODO",
+    "Painting Upgrade Tier II": "TODO",
+    "Belt Upgrade Tier VIII": "TODO",
+    "Miner Upgrade Tier VIII": "TODO",
+    "Processors Upgrade Tier VIII": "TODO",
+    "Painting Upgrade Tier VIII": "TODO",
+    "Belt Upgrade Tier M": "TODO",
+    "Miner Upgrade Tier M": "TODO",
+    "Processors Upgrade Tier M": "TODO",
+    "Painting Upgrade Tier M": "TODO"
+}
+
+categories = ["Belt", "Miner", "Processors", "Painting"]
+color_names = ["Red", "Blue", "Green", "Yellow", "Purple", "Cyan", "White", "Uncolored"]
+
+translate: List[Tuple[int, str]] = [
+    (1000, "M"),
+    (900, "CM"),
+    (500, "D"),
+    (400, "CD"),
+    (100, "C"),
+    (90, "XC"),
+    (50, "L"),
+    (40, "XL"),
+    (10, "X"),
+    (9, "IX"),
+    (5, "V"),
+    (4, "IV"),
+    (1, "I")
+]
+
+
+def roman(num: int) -> str:
+    """Converts positive integers into roman numbers."""
+    rom: str = ""
+    for key, val in translate:
+        while num >= key:
+            rom += val
+            num -= key
+    return rom
+
+
+def color_to_needed_building(color_list: List[str]) -> str:
+    for next_color in color_list:
+        if next_color in ["Yellow", "Purple", "Cyan", "White", "y", "p", "c", "w"]:
+            return "Mixed"
+    for next_color in color_list:
+        if next_color not in ["Uncolored", "u"]:
+            return "Painted"
+    return "Uncolored"
+
+
+shapesanity_simple: Dict[str, str] = {}
+shapesanity_1_4: Dict[str, str] = {}
+shapesanity_two_sided: Dict[str, str] = {}
+shapesanity_three_parts: Dict[str, str] = {}
+shapesanity_four_parts: Dict[str, str] = {}
+
+
+def init_shapesanity_pool() -> None:
+    # same shapes && same color
+    for color in color_names:
+        color_region = color_to_needed_building([color])
+        shapesanity_simple[f"{color} Circle"] = f"Shapesanity Full {color_region}"
+        shapesanity_simple[f"{color} Square"] = f"Shapesanity Full {color_region}"
+        shapesanity_simple[f"{color} Star"] = f"Shapesanity Full {color_region}"
+        shapesanity_simple[f"{color} Windmill"] = f"Shapesanity East Windmill {color_region}"
+    for shape in ["Circle", "Square", "Star", "Windmill"]:
+        for color in color_names:
+            color_region = color_to_needed_building([color])
+            shapesanity_simple[f"Half {color} {shape}"] = f"Shapesanity Half {color_region}"
+            shapesanity_simple[f"{color} {shape} Piece"] = f"Shapesanity Piece {color_region}"
+            shapesanity_simple[f"Cut Out {color} {shape}"] = f"Shapesanity Stitched {color_region}"
+            shapesanity_simple[f"Cornered {color} {shape}"] = f"Shapesanity Stitched {color_region}"
+    # one color && 4 shapes (including empty)
+    for first_color in ["r", "g", "b", "y", "p", "c"]:
+        for second_color in ["g", "b", "y", "p", "c", "w"]:
+            if not first_color == second_color:
```

Comment:
If you want to get rid of some indent depth here, a pattern I like is

```
for first_color in ["r", "g", "b", "y", "p", "c"]:
    for second_color in ["g", "b", "y", "p", "c", "w"]:
        if first_color == second_color:
            continue
        
        for third_color in ["b", "y", "p", "c", "w", "u"]:
            if third_color in [first_color, second_color]:
                continue
            
            ...
```

This reduces indent depth by one per condition that you guard-continue-clause like this